resources/uplink_set.rb

```diff
@@ -0,0 +1,123 @@
+# under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+# CONDITIONS OF ANY KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations under the License.
+
+OneviewCookbook::ResourceBaseProperties.load(self)
+
+property :native_network, String
+property :networks, Array
+property :fc_networks, Array
+property :fcoe_networks, Array
+property :logical_interconnect, String
+
+default_action :create
+
+action_class do
+  include OneviewCookbook::Helper
+  include OneviewCookbook::ResourceBase
+
+  def load_native_network(item)
+    if native_network
+      if native_network.nil? || native_network == 'nil'
+        item['nativeNetworkUri'] = nil
+      else
+        native_net = OneviewSDK::EthernetNetwork.find_by(item.client, name: native_network).first
+        raise "Network #{native_network} not found." unless native_net
+        item['nativeNetworkUri'] = native_net['uri']
+      end
+    end
+  end
+
+  def load_network(item)
+    if networks
+      if networks.empty?
+        item['networkUris'] = []
+      else
+        networks.each do |net_name|
+          net = OneviewSDK::EthernetNetwork.find_by(item.client, name: net_name).first
+          raise "Ethernet Network #{net_name} not found." unless net
+          item.add_network(net)
+        end
+      end
+    end
+  end
+
+  def load_fc_network(item)
+    if fc_networks
+      if fc_networks.empty?
+        item['fcNetworkUris'] = []
+      else
+        fc_networks.each do |net_name|
+          net = OneviewSDK::FCNetwork.find_by(item.client, name: net_name).first
+          raise "FC Network #{net_name} not found." unless net
+          item.add_fcnetwork(net)
+        end
+      end
+    end
+  end
+
+  def load_fcoe_network(item)
+    if fcoe_networks
+      if fcoe_networks.empty?
+        item['fcoeNetworkUris'] = []
+      else
+        fcoe_networks.each do |net_name|
+          net = OneviewSDK::FCoENetwork.find_by(item.client, name: net_name).first
+          raise "FCoE Network #{net_name} not found." unless net
+          item.add_fcoenetwork(net)
+        end
+      end
+    end
+  end
+
+  def load_logical_interconnect(item)
+    if logical_interconnect
+      li = OneviewSDK::LogicalInterconnect.find_by(item.client, name: logical_interconnect).first
+      raise "Logical Interconnect #{logical_interconnect} not found." unless li
+      item.set_logical_interconnect(li)
+    end
+  end
+
+  # Looks for enclosures within Location Entries
+  # An IF statement by itself would try to iterate and raise an exception if locationEntries does not exist
+  def load_enclosure(item)
+    if defined? item.data['portConfigInfos'][0]['location']['locationEntries']
+      item.data['portConfigInfos'][0]['location']['locationEntries'].each do |entry|
+        next unless entry && entry['type'] == 'Enclosure'
+        enclosure = OneviewSDK::Enclosure.find_by(item.client, name: entry['value']).first
```

Comment:
Typically this value is a URI. If they specify the URI instead of a name here, this will fail, won't it? Since the retrieve! method tries both, perhaps we can use that instead?